### PR TITLE
Add chinese(simplified/traditional) translate

### PR DESCRIPTION
**Pull request by SummonHIM**
And here has some recommend pixel font for chinese characters:
1. zpix (https://github.com/SolidZORO/zpix-pixel-font)
This font is free for **education** and **personal only**. support simplified, traditional and japan characters.
zpix

2. Chusung (https://diaowinner.itch.io/chusung)
This font is free for commercial, education and personal. support simplified and **some** traditional characters.
Chusung

3. IPix (https://purestudio.itch.io/ipix)
This font is free for commercial, education and personal. but only support simplified characters.
ipix

---
Update 1
localize config key, desc and warn
auto detect os language on first start

### CODE REVIEW

**Comment by OrsellGaming** [comment]
Hi @SummonHIM,

Thanks for your pull request for P2MM! This pull request is nice and fleshed out and you took time to localize the settings and even the test items, this looks amazing! I still need to add support for multiple fonts in order to be called apon for each language, but with you system language detector function it will be really easy to do! I would like to note though that on my fork I do have a few extra translations that aren't currently on dev branch. Do you mind after your pull request has been reviewed and accepted to help us translate any other new translations that we create and merge to dev?

Thanks a bunch!
Orsell

**Comment by SummonHIM** [comment]
> Hi @SummonHIM,
> 
> Thanks for your pull request for P2MM! This pull request is nice and fleshed out and you took time to localize the settings and even the test items, this looks amazing! I still need to add support for multiple fonts in order to be called apon for each language, but with you system language detector function it will be really easy to do! I would like to note though that on my fork I do have a few extra translations that aren't currently on dev branch. Do you mind after your pull request has been reviewed and accepted to help us translate any other new translations that we create and merge to dev?
> 
> Thanks a bunch! Orsell

sure, no problem.

File: src/Scripts/Configs.py [hunk 1]
```diff
@@ -9,6 +10,17 @@
...
+    sysDefaultLocale = locale.getdefaultlocale()[0]
+    if sysDefaultLocale.lower() == 'fr-FR'.lower() or sysDefaultLocale.lower() == 'fr-BE'.lower() or sysDefaultLocale.lower() == 'fr-CA'.lower() or sysDefaultLocale.lower() == 'fr-LU'.lower() or sysDefaultLocale.lower() == 'fr-CH'.lower():
+        return 'Fran\u00e7ais'
+    elif sysDefaultLocale.lower() == 'zh_CN'.lower() or sysDefaultLocale.lower() == 'zh-SG'.lower() or sysDefaultLocale.lower() == 'zh-Hans'.lower():
+        return 'SChinese'
+    elif sysDefaultLocale.lower() == 'zh-TW'.lower() or sysDefaultLocale.lower() == 'zh-HK'.lower() or sysDefaultLocale.lower() == 'zh-MO'.lower() or sysDefaultLocale == 'zh-Hant'.lower():
+        return 'TChinese'
+    else:
+        return "English"
+
```
Comment (cabiste69) [inline src/Scripts/Configs.py:23]: an easier and cleaner way to do this is to just get the system locale and only use the first 2 letters then just rename the languages files to 2 letters names `en.json`, `fr.json` etc... 
as for the chinese we can do a simple check like `if ("zh" in locale)  and ("tw" in locale) then return "TChinese"`

File: src/MainWindow.py [hunk 2]
```diff
@@ -312,11 +312,11 @@ def DefineResourcesMenu(self) -> None:
...
     def DefineTestingMenu(self) -> None:
         self.Button_InputField = self.ButtonTemplate(
-            "User Input", self.Button_InputField_func)
+            translations["test_user_input"], self.Button_InputField_func)
         self.PopupBox_gui = self.ButtonTemplate(
-            "Popup Box", self.PopupBox_test_func)
+            translations["test_popup_box"], self.PopupBox_test_func)
         self.Button_PrintToConsole = self.ButtonTemplate(
-            "Print to Console", self.Button_PrintToConsole_func)
+            translations["test_print_to_console"], self.Button_PrintToConsole_func)
```
Comment (cabiste69) [inline src/MainWindow.py:319]: for the "test buttons" i'd like if they stay english only so we don't need to add new values in the translation files every time we want to quickly debug something

**Comment by cabiste69** [comment]
> still need to add support for multiple fonts 

that shouldn't be an issue, just add a property in the json file with the location of the font

**Comment by SummonHIM** [comment]
the "test_" has removed

**Comment by cabiste69** [comment]
> the "test_" has removed

sorry but that's not what i meant by "remove the test", i meant the translation itself just keep them as they were before, also consider joining the discord to talk more about this

File: src/Scripts/Configs.py (hunk 1 above)
Comment (SummonHIM) [reply src/Scripts/Configs.py:29]: edited

**Comment by SummonHIM** [comment]
> > the "test_" has removed
> 
> sorry but that's not what i meant by "remove the test", i meant the translation itself just keep them as they were before, also consider joining the discord to talk more about this

i know. but i think keep the same naming convention will get less confusion for this project at first

**Comment by OrsellGaming** [comment]
What he means is that there is no point in having translations for the test menu and its items since all they are for is testing if things function correctly. We won't need translation for those essentially so it's fine just having the text be “Test Menu” and such.

**Comment by SummonHIM** [comment]
> What he means is that there is no point in having translations for the test menu and its items since all they are for is testing if things function correctly. We won't need translation for those essentially so it's fine just having the text be “Test Menu” and such.

got it. i restore it

**Comment by OrsellGaming** [comment]
> there have another issue is the desc size seem like little big in chinese
> 
> 
> 
> 屏幕截图 2023-01-10 235245
> 
> 

Now I'm unsure about this, I know the current dev branches launcher is buggy at the moment but a button shouldn't appear like this. What menu is this by the way?

**Comment by SummonHIM** [comment]
> > there have another issue is the desc size seem like little big in chinese
> > 屏幕截图 2023-01-10 235245
> 
> Now I'm unsure about this, I know the current dev branches launcher is buggy at the moment but a button shouldn't appear like this. What menu is this by the way?

Is the `setting/launcher config` page. the blue text is the description.

Or put a lot of space end of text to solve this issue?

**Comment by OrsellGaming** [comment]
Let's wait for the moment, I want to try and push my own fixes onto dev so it's nice and stabke, then if there are any issues I can tweak it and such. I also think we should use the zpix font since you said that not only does it support simple and traditional characters, but it also supports Japanese characters. So we could also get another supported language out of that. Is it possible for you to add at the top of the translation jsons a key for what font is used? Like the path to it?

**Comment by OrsellGaming** [comment]
image
Looks like this

**Comment by SummonHIM** [comment]
> image Looks like this, the path.

i replace all the path to `translate["font_path"]`. but the `class DisplayText` and the `class ButtonTemplate` has no defined the `translate` array. how to fix that?

my python skill are pretty sucks...

**Comment by OrsellGaming** [comment]
The screenshot I showed contained stuff that isn't currently on the dev branch, but I can easily add in what's needed once everything is merged. I think your pull request is ready I just need to fix some things on my end then I should be able to merge the pull request. Might take a some time though, we’ll see.

**Comment by OrsellGaming** [comment]
Hey @SummonHIM ! We have finally got dev stuff synced and merged! Feel free to sync your fork of the repo and make any changes you need. We'll look through the pull request once you've finished your part.

**Comment by SummonHIM** [comment]
> Hey @SummonHIM ! We have finally got dev stuff synced and merged! Feel free to sync your fork of the repo and make any changes you need. We'll look through the pull request once you've finished your part.

Finished! It can merge anytime.

**Comment by SummonHIM** [comment]
> Hey @SummonHIM ! We have finally got dev stuff synced and merged! Feel free to sync your fork of the repo and make any changes you need. We'll look through the pull request once you've finished your part.

The translate is finished.

**Comment by OrsellGaming** [comment]
Hey @SummonHIM ! Can you add the zpix font into the fonts folder for the launcher? That way we can use it in the launcher, could you also add the "font" key to the top of each language json too? Thanks!

**Comment by SummonHIM** [comment]
> Hey @SummonHIM ! Can you add the zpix font into the fonts folder for the launcher? That way we can use it in the launcher, could you also add the "font" key to the top of each language json too? Thanks!

Added

**Comment by OrsellGaming** [comment]
Hey @SummonHIM ! Is it possible for you to do a translation for the `README.md` file? It's possible to make one that just appears for just those who have simplified/traditional chinese set as their primary language by naming the file `README.(The language code for traditional/simplified chinese).md`. It would be greatly appreciated!

**Comment by SummonHIM** [comment]
> Hey @SummonHIM ! Is it possible for you to do a translation for the `README.md` file? It's possible to make one that just appears for just those who have simplified/traditional chinese set as their primary language by naming the file `README.(The language code for traditional/simplified chinese).md`. It would be greatly appreciated!

Sure! uploaded

File: README/README.zh-CN.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <h1>
-  <img src="Images/p2mmbanner_logo.png" alt="p2mmbanner_logo" width="500" height="300" align="left">
+  <img src="../Images/p2mmbanner_logo.png" alt="p2mmbanner_logo" width="500" height="300" align="left">
```
Comment (OrsellGaming) [inline README/README.zh-CN.md:2]: May I ask what this `../` is for for the path to the logo?

Comment (OrsellGaming) [reply README/README.zh-CN.md:2]: Why is this needed exactly?

**Review by OrsellGaming — COMMENTED** [review]
May I ask what this ../ is for for the path to the logo?

File: README/README.zh-CN.md (hunk 3 above)
Comment (SummonHIM) [reply README/README.zh-CN.md:2]: Because i storage the file in ”Readme“ folder. need `../` to call the logo.
or just storage in root will better?

Comment (OrsellGaming) [reply README/README.zh-CN.md:2]: Normally it already targets the logo just fine without `../`, normally it starts at root and looks for it from there. But I didn't know this also worked.

**Pull request merged**